File: Dataset/Leetcode/valid/55/623.java
 class Solution {
    public boolean XXX(int[] nums) {
        int farest = 0; //表示最远可以跳到的位置
        for(int i = 0;i <= farest && i < nums.length;i++){
            farest = Math.max(farest,i+nums[i]);
        }
        return farest>= (nums.length-1);
    }
}

File: Dataset/Leetcode/train/70/623.cpp
class Solution {
public:
    // int XXX(int n) {
    //     if(n == 1)
    //         return 1;
    //     if(n == 2)
    //         return 2;
    //     return XXX(n-1) + XXX(n-2);
    // }
    int XXX(int n) {
        if(n == 1)
            return 1;
        if(n == 2)
            return 2;
        int f1 = 1;
        int f2 = 2;
        int fn = 0;
        for(int i=3; i<=n; i++){
            fn = f1 + f2;
            f1 = f2;
            f2 = fn;
        }
        return fn;
    }
};

